objectivec/ort_value.mm

```diff
@@ -127,6 +155,29 @@ - (nullable NSMutableData*)tensorDataWithError:(NSError**)error {
   ORT_OBJC_API_IMPL_CATCH_RETURNING_NULLABLE(error)
 }
 
+- (nullable NSArray<NSString*>*)tensorStringDataWithError:(NSError**)error {
+  try {
+    const auto tensorTypeAndShapeInfo = _typeInfo->GetTensorTypeAndShapeInfo();
+    const size_t elementCount = tensorTypeAndShapeInfo.GetElementCount();
+    const size_t tensorStringDataLength = _value->GetStringTensorDataLength();
+    std::string tensorStringData(tensorStringDataLength, '\0');
+    std::vector<size_t> offsets(elementCount);
+    _value->GetStringTensorContent(const_cast<char*>(tensorStringData.data()), tensorStringDataLength,
```

Comment:
I think we don't need `const_cast` as data() has an overload returning a non-const pointer. also maybe we can just use `vector<char>`. `std::string` has a trailing null which we don't care about here.